engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -241,7 +242,7 @@ public LogChannelInterface getLogChannel() {
     return log;
   }
 
-  public String getXML() {
+  public synchronized String getXML() {
```

Comment:
A ReadWrite Lock can accommodate this just fine. If a Read lock is in effect the write will wait. If a Write is in affect the read will wait. Concurrent reads are not mutexed